tests/unit/test_charm.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 dima.tisnek@canonical.com
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+import pytest
+from ops.testing import Harness
+from ops.model import ActiveStatus, MaintenanceStatus
+from charm import HexanatorCharm
+
+
+@pytest.fixture
+def harness():
+    harness = Harness(HexanatorCharm)
+    harness.begin()
+    yield harness
+    harness.cleanup()
+
+
+def test_pebble_ready(harness):
+    assert harness.model.unit.status == MaintenanceStatus('')
+
+    harness.container_pebble_ready('gubernator')
+
+    assert harness.model.unit.status == ActiveStatus()
```

Comment:
The test pebble client actually records something... except in this case, it's not configured with any services, so doesn't record anything. In a way it's a complication stemming from pebble config being backed into rockcraft.yaml, and unit test doesn't know what's in the container [image].

I'll have to think of something for this!

Reply:
Well, it sets the service status but you have to first set it to something else, because you can't tell if a service was 'restarted', only that is ends up running.

So you could do harness.charm.unit.containers['...'].add_layer() and then harness.charm.unit.containers['...'].stop(...) as part of the setup. It's a bit clunky, but as far as I know, Harness was designed for charms dynamically adding Pebble layers, not for them to be coming from the image.

There are plans ([e.g.](https://github.com/canonical/ops-scenario/issues/34)) to properly model service status in Scenario, but that's unlikely to be in 7.0.

Reply:
After discussion at GitHub issues, I'll defer that to:

https://github.com/canonical/ops-scenario/issues/152
https://github.com/canonical/ops-scenario/issues/34